pkg/storage/ingest/fetcher.go

```diff
@@ -334,7 +339,9 @@ func newConcurrentFetchers(
 		metrics:                 metrics,
 		minBytesWaitTime:        minBytesWaitTime,
 		lastReturnedOffset:      startOffset - 1,
-		startOffsets:            startOffsetsReader,
+		startOffset:             startOffset,
+		startConcurrency:        concurrency,
```

Comment:
To reduce signature changes, I preferred to store the `startOffset` and `startConcurrency` on the `concurrentFetchers` struct, and then just read them in the new `Start()` function. `startOffset` gets manipulated in this function too.